src/input.rs

```diff
@@ -365,24 +365,35 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     }
 
     pub fn on_mouse_wheel(&mut self, delta: MouseScrollDelta, phase: TouchPhase) {
-        let modes = mode::MOUSE_REPORT_CLICK | mode::MOUSE_MOTION | mode::SGR_MOUSE;
+        let modes = mode::MOUSE_REPORT_CLICK | mode::MOUSE_MOTION | mode::SGR_MOUSE | mode::ALT_SCREEN_BUF;
         if !self.ctx.terminal_mode().intersects(modes) {
             return;
         }
 
         match delta {
             MouseScrollDelta::LineDelta(_columns, lines) => {
                 let to_scroll = self.ctx.mouse_mut().lines_scrolled + lines;
-
                 let code = if to_scroll > 0.0 {
                     64
                 } else {
                     65
                 };
 
                 for _ in 0..(to_scroll.abs() as usize) {
-                    self.normal_mouse_report(code);
+                    if self.ctx.terminal_mode().intersects(mode::ALT_SCREEN_BUF) {
+                        // Faux scrolling
+                        if code == 64 {
+                            // Scroll up one line
+                            self.ctx.write_to_pty("\x1bOA".as_bytes());
+                        } else {
+                            // Scroll down one line
+                            self.ctx.write_to_pty("\x1bOB".as_bytes());
```

Comment:
>Whenever the user scrolls up or down while the alternate screen buffer is active, instead of actual scrolling three up/down arrow keys are inserted.

Isn't it currently one up/down arrow key? 

This patch works great for me on macOS 10.13.

Reply:
Yes that's correct. Initially I went for three scrolls but I like the behavior a lot more with just one. Because of that I changed that behavior in a later commit.

Reply:
@chrisduerr What about making it configurable?